### ISSUE TITLE
New rule: `component-options-name-casing`

### ISSUE THREAD

**Issue by g-plane** [vuejs/eslint-plugin-vue#1723]
**Please describe what the rule should do:**

Enforce the casing of component name in `components` options.

This rule should allow two options: `PascalCase` and `camelCase`, and default option is `PascalCase`.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script>
import myComponent from './myComponent.vue'

export default {
  components: {
    myComponent  // <-- should be `MyComponent`
  }
}
</script>
```

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for rule proposal!

I don't know if any users like camelCase, but the rule of checking the casing sounds good to me :+1:
